event_response_tool/event_response.py

```diff
@@ -0,0 +1,323 @@
+                or len(event_response_inputs.old_analysis_profile_uuid) == 0
+            )
+        ):
+            alert.error(
+                "Please provide inputs for either duplicating LayerViews or creating/updating Event Response Analysis Profile"
+            )
+
+
+def check_and_set_optional_inputs(event_response_inputs):
+    event_response_inputs.analyzere_url = (
+        event_response_inputs.analyzere_url
+        if event_response_inputs.analyzere_url
+        else config.get("server", "base_url")
+    )
+    event_response_inputs.analyzere_username = (
+        event_response_inputs.analyzere_username
+        if event_response_inputs.analyzere_username
+        else config.get("server", "username")
+    )
+    event_response_inputs.analyzere_password = (
+        event_response_inputs.analyzere_password
+        if event_response_inputs.analyzere_password
+        else config.get("server", "password")
+    )
+    event_response_inputs.catalog_description = (
+        event_response_inputs.catalog_description
+        if event_response_inputs.catalog_description
+        else config.get("ap_creator", "default_catalog_name")
+    )
+    event_response_inputs.simulation_description = (
+        event_response_inputs.simulation_description
+        if event_response_inputs.simulation_description
+        else config.get("ap_creator", "default_simulation_name")
+    )
+    event_response_inputs.analysis_profile_description = (
+        event_response_inputs.analysis_profile_description
+        if event_response_inputs.analysis_profile_description
+        else config.get("ap_creator", "default_analysis_profile_name")
+    )
+
+    return event_response_inputs
+
+
+def process(event_response_inputs):
+    # Accepted attributes
+    #     analyzere_url - Analyze Re URL
+    #     analyzere_username - Analyze Re username
+    #     analyzere_password - Analyze Re password
+    #     event_weights_csv - The CSV containing event weights
+    #     layer_views_csv - The CSV containing list of layer_views UUID
+    #     portfolio_view_uuid - The UUID of the portfolio_view (from where layer_views can be retrieved)
+    #     total_number_of_events - The total number of events to be created in the event response analysis profile
+    #     trial_count - The maximum number of trial per event
+    #     catalog_description - The description of the new event catalog to be created
+    #     simulation_description - The description of the new simgrid to be created
+    #     analysis_profile_description - The description of the new analysis profile to be created
+    #     old_analysis_profile_uuid - The UUID of the analysis profile to be updated with a new simgrid
+
+    warnings.filterwarnings("ignore")
+    init_logging()
+    output_dir = init_directories()
+
+    # Validate inputs
+    validate_inputs(event_response_inputs)
+    event_response_inputs = check_and_set_optional_inputs(
```

Comment:
should this not happen before `validate_inputs`?

Reply:
Agreed. Done now.